skycatalogs/objects/snana_object.py

```diff
@@ -0,0 +1,205 @@
+import galsim
+import h5py
+import numpy as np
+from .base_object import BaseObject,ObjectCollection
+
+__all__ = ['SnanaObject', 'SnanaCollection']
+
+class SnanaObject(BaseObject):
+    _type_name = 'snana'
+
+    def __init__(self, ra, dec, id, object_type, belongs_to, belongs_index,
+                 redshift=None):
+        super().__init__(ra, dec, id, object_type, belongs_to, belongs_index,
+                         redshift)
+        self._mjds = None
+        self._lambda = None
+
+        # indices of elements in mjd array bounding our mjd
+        self._mjd_ix_l = None
+        self._mjd_ix_u = None
+
+    def _get_sed(self, mjd=None):
+        if mjd is None:
+            mjd = self._belongs_to._mjd
+        mjd_start = self.get_native_attribute('start_mjd')
+        mjd_end = self.get_native_attribute('end_mjd')
+        if mjd < mjd_start or mjd > mjd_end:
+            return None, 0.0
+
+        return self._linear_interp_SED(mjd), 0.0
+
+    def get_gsobject_components(self, gsparams=None, rng=None):
+        if gsparams is not None:
+            gsparams = galsim.GSParams(**gsparams)
+        return {'this_object': galsim.DeltaFunction(gsparams=gsparams)}
+
+    def get_observer_sed_component(self, component, mjd=None):
+        sed, _ = self._get_sed(mjd=mjd)
+        if sed is not None:
+            sed = self._apply_component_extinction(sed)
+        return sed
+
+    def get_LSST_flux(self, bandpass, sed=None, cache=True, mjd=None):
+        def _to_flux(m):
+            return np.exp(-0.921034037196184 * m)
```

Comment:
This form was copied from similar expressions in older code, which I assumed used it because it was more efficient.  I can add the same explanatory comment here.